### DISCUSSION TITLE
Difference between `init powershell` and `init pwsh`

### DISCUSSION THREAD

**Discussion by mattcargile** [JanDeDobbeleer/oh-my-posh#2075] [Troubleshoot] [answered]
I noticed these differences in the code but see `init pwsh` preferred in the Docs. ~Is there a current or future difference for Powershell Core vs Windows Powershell , assuming that is the main difference and it isn't only an alias.~ So I do see more differences now and I wrote an `if` block in my `$PROFILE`. I guess the idea is we should potentially have different `omp.ps1` files or different code paths depending on which shell. So far `init pwsh` has worked for both `powershell.exe` and `pwsh.exe`. 

I guess what I saying is maybe it makes sense to add another entry distinguishing between `pwsh` and `powershell` in the docs.

**Comment by JanDeDobbeleer** [comment, marked as answer]
This was necessary before the refactoring of the init CLI. Nowadays they can both be the same.

**Reply by mattcargile** [reply]
So the intent is to support `powershell.exe` for the foreseeable future and I don't need to have code like the below in my `$PROFILE` to support both `powershell.exe` and `pwsh.exe`?

```powershell
if ($IsCoreCLR) { oh-my-posh.exe init pwsh --config $OhMyPoshConfig | Invoke-Expression }
else { oh-my-posh.exe init powershell --config $OhMyPoshConfig | Invoke-Expression }
```

**Reply by mattcargile** [reply]
I guess because I could see a world where the `omp.ps1` bootstrap script could diverge in Powershell Core vs Windows Powershell.

**Reply by JanDeDobbeleer** [reply]
They don't yet. I'll check but I don't think they do something different today.
